### PR TITLE
[READY] Adds alpha masking helper, which cryo now uses

### LINKED ISSUE #40922: Lizard tails stick out of cryo pods
lizzer tail

my liz build: Tail Light Tiger, Snout Sharp + Light, Horns Angeler Frills, Spines Short + Membrane, Body Markings Light Belly, Legs Digitigrade (but those don't work currently)

### PR DESCRIPTION
**Pull request by nicbn**
Fixes #40922

## Changelog
:cl:
fix: Icons will no longer extend past cryo.
imagedel: Xenos and monkeys no longer have snowflake icons for cryo.
/:cl:

### CODE REVIEW

**Comment by MrPerson** [comment]
At some point in the future, someone really oughta add a helper for alpha masking like this.

**Comment by nicbn** [comment]
Currently the machine I'm using right now has no support for BLEND_MULTIPLY, so I can't test it, but it should work.
This is DNM until I (or someone else) test it.

**Comment by nicbn** [comment]
Tested and good to go

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 1]
```diff
@@ -146,20 +138,30 @@
...
+	if(overlay_pixel_y != 1) // Same effect as overlay_pixel_y == 0 || occupant_overlay.pixel_y == 2
+		anim_up = overlay_pixel_y == 0 // Same effect as if(overlay_pixel_y == 0) anim_up = TRUE ; if(overlay_pixel_y == 2) anim_up = FALSE
 	if(anim_up)
-		occupant_overlay.pixel_y++
+		overlay_pixel_y++
 	else
-		occupant_overlay.pixel_y--
-	add_overlay(occupant_overlay)
+		overlay_pixel_y--
+	add_masked_overlay(is32x32, occupant_overlay)
```
Comment (AnturK) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: Can you use looped animate here instead of reapplying ?

Comment (nicbn) [reply code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: `animate` doesn't work for overlays

Comment (AnturK) [reply code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: You can use `vis_contents` (technically also images but that's not the best use case here)

Comment (nicbn) [reply code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: It is impossible to use blend mode with `vis_contents`, so it wouldn't work for non-32x32.

**Comment by nicbn** [comment]
According to Lummox, on 513 there may be built in alpha masking. That way we'll be able to use `animate` without having a `vis_contents` mess. If 513 comes out without alpha masking I may go back to this and try the `vis_contents` approach.
Meanwhile, the timers will do the trick.

**Comment by optimumtact** [comment]
I think alpha masking was confirmed, can you add an ifdef version check and crash the compile so we don't forget to update this?

**Pull request merged**